src/newlines.cpp

```diff
@@ -2234,6 +2234,13 @@ static bool collapse_empty_body(Chunk *br_open)
    {
       return(false);
    }
+   log_rule_B("nl_collapse_empty_body_ignore_functions");
+
+   if (  options::nl_collapse_empty_body_ignore_functions()
+      && br_open->GetPrevNcNnlNet()->Is(CT_FPAREN_CLOSE))
```

Comment:
I am confused here. Why are we looking back from the open brace? Shouldn't we look forward? What am I missing?

Reply:
Maybe it's possibly to do it in a simpler/more straightforward way, if so please tell me :)

Basically I want to check whether the brace belongs to a function definition since this rule is only meant to be triggered for function braces. This checks if the previous chunk is a closing parenthesis that belongs to a function (I think so at least).

Reply:
You can use the chunk parent to check if it s a function, like this:
```
# Line                Tag         Parent_type  Type of the parent         Columns Br/Lvl/pp         Flag   Nl  Text
#   1>               TYPE|           FUNC_DEF|     PARENT_NOT_SET[  1/  1/  5/  0][0/0/0][        c 0000][0-0] void
#   1>           FUNC_DEF|               NONE|     PARENT_NOT_SET[  6/  6/  9/  1][0/0/0][             0][0-0]      bar
#   1>        FPAREN_OPEN|           FUNC_DEF|     PARENT_NOT_SET[  9/  9/ 10/  0][0/0/0][   2 0000 0000][0-0]         (
#   1>               TYPE|               NONE|     PARENT_NOT_SET[ 10/ 10/ 14/  0][0/1/0][        8 0008][0-0]          void
#   1>       FPAREN_CLOSE|           FUNC_DEF|     PARENT_NOT_SET[ 14/ 14/ 15/  0][0/0/0][   2 0000 0008][0-0]              )
#   1>            NEWLINE|               NONE|     PARENT_NOT_SET[ 15/ 15/  1/  0][0/0/0][             0][1-0]
#   2>         BRACE_OPEN|           FUNC_DEF|     PARENT_NOT_SET[  1/  1/  2/  0][0/0/0][   2 0000 0000][0-0] {
#   2>            NEWLINE|               NONE|     PARENT_NOT_SET[  2/  2/  8/  0][1/1/0][             0][1-0]
#   3>                 IF|               NONE|     PARENT_NOT_SET[  8/  8/ 10/  0][1/1/0][        c 0000][0-0]        if
```
You can see the brace open parent is ```FUNC_DEF```.
You can use the ```-ds dump``` option on a test file to get a list of files showing you all the chunks during the process, this will help you finding more info.

Btw, could you provide an example of what the option should and should not do? I see one expected positive test but not a negative one, so I don't get the difference yet

Reply:
Oh yeah, checking the parent is way easier. Thanks :)

Good point about the tests. I've added a few more examples. I guess the "ignore_functions" isn't technically correct as it expands/uncollapses(?) the braces, but I couldn't come up with a more correct name that wasn't obnoxiously long.